src/services/photos/sync/devicePhotosScanner/devicePhotosScanner.ts

```diff
@@ -118,7 +118,10 @@ export class DevicePhotosScannerService extends RunnableService<DevicePhotosScan
 
   private saveCachedDevicePhotos(assets: MediaLibrary.Asset[]) {
     assets.forEach((asset) => {
-      const name = asset.filename.split('.')[0];
+      let name = asset.filename;
+      if (asset.filename.includes('.')) {
+        name = asset.filename.split('.')[0];
```

Comment:
Be careful with photos whose name has been edited as the name should be from the beginning until before the last point that exists on the string

Reply:
this filename is the one that the library provides us (expo-media-library). On iOS the filename cannot be edited since the images doesn't have filename at all, only an ID, (expo-media-library generates a consistent name from that).

On Android, since the filesystem is basically open, we only get the items that the library provides us based on our filters (we ask for photos and videos). I did a test and if you rename a file on Android from .jpg to .mp3 the item no longer is picked by expo-media-library so we are safe there.